modules/AWS/rds/main.tf

```diff
@@ -37,6 +37,7 @@ module "db" {
   major_engine_version = var.major_engine_version # DB option group
   instance_class       = var.instance_class
 
+  allow_major_version_upgrade = true
```

Comment:
I added a flag in configuration file to make this controlled by the user and keep the default as `false`. However, on the second thought we may trade the simplicity and avoid the fine tuned config which makes the configuration a bit more complex. If the team thinks we should keep this simple, I can close the other PR that I drafted to handle this flag.

Reply:
Adding this is similar to the v14 postgres discussion. Why add this if its not needed or going to be used? My understanding from the vendors perspective is that this is something the are not likely to do. Although I could be wrong, might be worth getting the product owners take on this.

Reply:
Who is the best person to contact?

Reply:
I can raise a question for `dc-app` team. If we not add this flag then we must mention that RDS version cannot be changed after first install in our documentation.

Reply:
@nghazali  any update?

Reply:
The fist option is okay with them.... I mean we can skip this part for now.

Reply:
I just trying to test restoring data from snapshot and it looks we need to keep `AllowMajorVersionUpgrade` flag otherwise we might get this error:
```
Error: error modifying DB Instance (atlas-jira-blitz-snapshot-test-jira-db): InvalidParameterCombination: The AllowMajorVersionUpgrade flag must be present when upgrading to a new major version.
	status code: 400, request id: 94a311e8-862a-42d8-b200-45e6ef034031
```

Reply:
This is because the snapshot might required DB engine upgrade based on the snapshot engine version and the major version in the configuration file.

Reply:
Well we could avoid this is we make sure the engine version in the `config.tfvars` is the same as the snapshot `11`. Remember the DCAPT will supply the appropriate `config.tfvars` so we shouldnt need to worry about this. Its a nice fallback option but I just wonder if its really necessary to upgrade the snapshot from `11` to the one specified i.e. `<product>_db_major_engine_version` when we could just alter `<product>_db_major_engine_version` to match `11`

Reply:
This is removed in this PR we can talk further in  https://github.com/atlassian-labs/data-center-terraform/pull/193